test/blackbox-tests/test-cases/directory-targets/create-target-chdir.t

```diff
@@ -16,16 +16,10 @@ Attempt to create a directory with chdir + with-stdout-to:
   > EOF
 
   $ dune build foobar/
-  File "dune", line 1, characters 0-130:
-  1 | (rule
-  2 |  (targets (dir output))
-  3 |  (deps (sandbox always))
-  4 |  (action
-  5 |   (progn
-  6 |    (chdir output
+  File "dune", line 7, characters 20-21:
   7 |     (with-stdout-to x (echo foobar))))))
-  Error: Rule has targets in different directories.
```

Comment:
I think we should try to change the test, so we still test this other code path.

Reply:
I prefer the new error too, of course.

I'm just worried about no longer testing the other code path. If, as you say, it is no longer reachable, perhaps we should prune it away?

Reply:
It can still happen under the circumstance of us using `Super_context.add_rule` manually. However, in that case, that would be a bug with our code calling this function incorrectly. So I suppose we can consider changing it into a Code_error?

Reply:
ping @snowleopard

Reply:
(Hmm, I'm pretty sure I wrote a response here, but it's not here!)

If you are confident that the only way to trigger the old error is via `add_rule` then yes, switching to code error seems best.

Reply:
I switched it